### ISSUE TITLE
Single-FOV wells vs. multi-fov wells

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal#74]
We're [building ome-zarr reader support for multi-fov wells](https://github.com/fractal-analytics-platform/mwe_fractal/issues/66), where we would save each fov/site from the microscope as its own nested directory in the OME-Zarr file.

While building this and while talking to Kevin Yamauchi, some thoughts came up on pros and cons of the different strategies and I'll try to summarize them here in this high level issue, so that we can keep an overview here.

----------------------------

### Definitions:
**Single-FOV wells:** Describes a way of saving image data to OME-Zarr in which data for each well (each unit of a plate) is saved in a single field of view.

Single FOV schematic adapted from OME-NGFF:
```
5966.zarr                 # One plate (id=5966) converted to Zarr
    ├── .zgroup
    ├── .zattrs               # Implements "plate" specification
    ├── A                     # First row of the plate
    │   ├── .zgroup
    │   │
    │   ├── 1                 # First column of row A
    │   │   ├── .zgroup
    │   │   ├── .zattrs       # Implements "well" specification
    │   │   │
    │   │   └── 0             # Single field of view of well A1
    │   │       │
    │   │       ├── .zgroup
    │   │       ├── .zattrs   # Implements "multiscales", "omero"
    │   │       ├── 0
    │   │       │   ...       # Resolution levels
    │   │       ├── n
    │   │       └── labels    # Labels (optional)
    │   ├── ...               # Columns
    │   └── 12
    ├── ...                   # Rows
    └── H
```

**Multi-FOV wells:** Describes a way of saving image data to OME-Zarr in which data for each original image acquisition region (=field of view, site) is saved to a separate folder under the well (unit of a plate) folder.

```
5966.zarr                 # One plate (id=5966) converted to Zarr
    ├── .zgroup
    ├── .zattrs               # Implements "plate" specification
    ├── A                     # First row of the plate
    │   ├── .zgroup
    │   │
    │   ├── 1                 # First column of row A
    │   │   ├── .zgroup
    │   │   ├── .zattrs       # Implements "well" specification
    │   │   │
    │   │   ├── 0             # First field of view of well A1
    │   │   │   │
    │   │   │   ├── .zgroup
    │   │   │   ├── .zattrs   # Implements "multiscales", "omero"
    │   │   │   ├── 0
    │   │   │   │   ...       # Resolution levels
    │   │   │   ├── n
    │   │   │   └── labels    # Labels (optional)
    │   │   ├── ...           # Multiple fields of views, saved separately
    │   │   └── m             # Last field of view (e.g. in a well with 6 rows & 7 columns, fov 41)
    │   ├── ...               # Columns
    │   └── 12
    ├── ...                   # Rows
    └── H
```

---------------------------

### Benefits of using multi-FOV approach
- This follows the OME-NGFF v0.4 (& v0.5-dev) standard
- Makes for an easy parallelization scheme: The same way we parallelize over plates and wells, we could also parallelize over sites. Makes it easy and robust to run illumination correction per site, run image analysis per site etc.
- Easily understandable on where original images go => the folder of the site
- Allows a user to load a single site or a few sites (and if the metadata is parsed, those sites are placed correctly, which [works in our test cases](https://github.com/fractal-analytics-platform/mwe_fractal/issues/66#issuecomment-1157845418))
- Easy support for search-first data: We only save the fovs & their position, the reader handles placement

### Concerns with multi-FOV approach
- How will read performance in the napari viewer be? Given that we have everything saved per site again, the pyramids are also built by site. That means we can't chunk pyramids across sites anymore, thus have the same amounts of files per level, with files just becoming smaller. Improving chunking was one of the things we changed in May that improved viewer performance (see here: https://github.com/fractal-analytics-platform/mwe_fractal/issues/32)
- With increased pyramid files, the number of overall files also increase, thus increasing the burden on IT infrastructure if we don't use an object storage approach or Zarr zipping (see here: https://github.com/fractal-analytics-platform/mwe_fractal/issues/59)

### Benefits of single-FOV approach
- Already works with the current OME-Zarr napari reader plugin
- "Fast" / decent visualization speeds
- Single-FOV setups would be nice to display stitched data (though what would we do before stitching? Tile them next to each other and just have an image of a different size after stitching?)

### Concerns with single-FOV approach
- How do we handle parallelization per site? We could base it on chunk size at the lowest level (but on some level, that is something that could always be changed or that we'd want to optimize for either processing or visualization)
- Parallelization: If we process in smaller chunks, but write to an AnnData object that is saved "per site" (and now there is only 1 site) => How do we handle parallel write access to that file? Or how do we work around having to write to it in parallel?
- How do we handle site-wise label images if all sites are fused? Or do we process per site and then find a way to relabel once the whole well is processed? (we wouldn't want multiple objects with "label" = 1 in the same label image)
=> This opens the larger question though: How will we handle "region of interest" based computation, e.g. for an organoid that crosses site boundaries or for multiple sites that we want to stitch together and then process as one?

---------------------------

We will test some of the concerns regarding performance with the [planned work on the ome-zarr-py plugin](https://github.com/fractal-analytics-platform/mwe_fractal/issues/66). Once we have a better understanding of its performance implications, we should know better how to judge the trade-offs illustrated above.

**Comment by jluethi** [comment]
It looks like parallelization over ROIs works well (https://github.com/fractal-analytics-platform/fractal/issues/95) and the visualization scales well with this new approach.

The only downside to the multi-FOV approach seems to be handling overlap between images. We have a new issue specifically for this [here](https://github.com/fractal-analytics-platform/fractal/issues/120), so let's close this one.
